osc-server/src/main/java/org/osc/core/server/Server.java

```diff
@@ -550,8 +566,16 @@ public void shutdownRabbitMq() {
             // No problem - this means the service was
             // already unregistered (e.g. by bundle stop)
         }
-        this.deploymentSpecRunnerCSO.ungetService(this.rabbitMQRunner.getOsDeploymentSpecNotificationRunner());
-        this.securityGroupRunnerCSO.ungetService(this.rabbitMQRunner.getSecurityGroupRunner());
+        ComponentServiceObjects<OsDeploymentSpecNotificationRunner> cso = this.deploymentSpecRunnerCSO;
+        if(cso != null) {
```

Comment:
nitpick, i would just use deploymentspecrunnerCSO instead of calling it cso and cso2